clientv3/op.go

```diff
@@ -77,6 +77,13 @@ func (op Op) toRequestOp() *pb.RequestOp {
 			r.SortOrder = pb.RangeRequest_SortOrder(op.sort.Order)
 			r.SortTarget = pb.RangeRequest_SortTarget(op.sort.Target)
 		}
+		// By default, mvcc.Range returns results already sorted by keys in lexiographically
+		// ascending order. When SortOrder is specified, server fetches the entire key-space
+		// first, and then applies sorting or limits. For key-ascending-ordered results with limit,
+		// client should ignore the sort option to avoid this unnecessary workload.
+		if r.Limit > 0 && r.SortOrder == pb.RangeRequest_ASCEND && r.SortTarget == pb.RangeRequest_KEY {
```

Comment:
can this go in `WithSort`? I don't think the limit matters for this optimization.